tests/pages/contribute/signup.py

```diff
@@ -0,0 +1,89 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.select import Select
+from selenium.webdriver.support.ui import WebDriverWait as Wait
+
+from .base import ContributeBasePage
+
+
+class ContributeSignUpPage(ContributeBasePage):
+
+    _url = '{base_url}/{locale}/contribute/signup'
+
+    _name_locator = (By.ID, 'id_name')
+    _email_locator = (By.ID, 'id_email')
+    _country_locator = (By.ID, 'id_country')
+    _message_locator = (By.ID, 'person-message')
+    _html_format_locator = (By.ID, 'id_format_0')
+    _text_format_locator = (By.ID, 'id_format_1')
+    _privacy_policy_checkbox_locator = (By.ID, 'id_privacy')
+    _newsletter_checkbox_locator = (By.ID, 'id_newsletter')
+    _submit_button_locator = (By.CSS_SELECTOR, '.submit > button')
+    _success_url_slug = '/contribute/thankyou/'
+    _areas_region_locator = (By.CLASS_NAME, 'areas')
+    _coding_category_locator = (By.CLASS_NAME, 'category-coding')
+    _coding_area_locator = (By.CSS_SELECTOR, '#area-coding select')
+    _helping_category_locator = (By.CLASS_NAME, 'category-helping')
+
+    def select_coding_category(self):
+        self.selenium.find_element(*self._coding_category_locator).click()
+        self.wait_for_coding_area_to_display()
+
+    def select_helping_category(self):
+        self.selenium.find_element(*self._helping_category_locator).click()
+
+    def wait_for_coding_area_to_display(self):
+        # required attribute is set once the area finishes animating
+        assert not self.is_coding_area_required, 'Coding area is already required'
```

Comment:
This looks like it could cause a race condition. I don't think we need `wait_for_coding_area_to_display` or the initial check for the coding area to be not required. The wait can go directly into `select_coding_category` as it's only called from there anyway.